### DISCUSSION TITLE
Failed to cancel request Error at long query

### DISCUSSION THREAD

**Discussion by koneubauer** [Budibase/budibase#11360] [Help]
We want to run a query which normally takes 90 seconds to return results. Whe have set the QUERY_THREAD_TIMEOUT variable to 180000. However, when running the query, we get the following error: 
![image](https://github.com/Budibase/budibase/assets/117980610/98a9823d-8000-4f12-bd63-354530916157)
Is there another env-variable for this?
Running a docker-compose v2.8.10

**Comment by ConorWebb96** [comment]
Hey @koneubauer,

You should be able to include this within your `.env` file. An example of this can be found here; https://github.com/Budibase/budibase/issues/6078#issuecomment-1139049852. Make sure after adding the variable to your env file that you restart your docker containers.

I hope this helps.

**Comment by koneubauer** [comment]
Hey @ConorWebb96,

thanks. We already set the QUERY_THREAD_TIMEOUT in the .env file as seen in #3891 . Now the above error appears. Before we set the query-timeout to 180000 this error appeared: 
![image](https://github.com/Budibase/budibase/assets/117980610/cbcc1ef1-0b0a-4576-8536-699b83a5abc4)

**Reply by ConorWebb96** [reply]
Hey @koneubauer,

Have you definitely restarted your containers? I've just run through it from start to finish and was able to change the query time outs. Using a combination of my `.env` file and `docker-compose.yaml`

**docker-compose.yaml**
<img width="716" alt="Screenshot 2023-07-27 at 14 34 24" src="https://github.com/Budibase/budibase/assets/126772285/6da7fa8b-e102-40a4-b618-662f6fc73014">

**.env variable**
<img width="334" alt="Screenshot 2023-07-27 at 14 28 05" src="https://github.com/Budibase/budibase/assets/126772285/ba35d703-4966-4693-84da-00f7c942410c">

**End result**
<img width="1066" alt="Screenshot 2023-07-27 at 14 36 37" src="https://github.com/Budibase/budibase/assets/126772285/382e6f08-d5ba-46a6-a649-dce8c5299a1b">

I hope this helps.

**Reply by koneubauer** [reply]
Hey,

which container exactly should we restart? And is a restart enough or do we have to run compose down and compose up?

**Reply by ConorWebb96** [reply]
Hey @koneubauer,

What version of self-host are you using?

**Reply by koneubauer** [reply]
We're currently using 2.9.5

**Reply by ConorWebb96** [reply]
Sorry I meant are you using docker single image, docker compose?

I've re-read the chat you are using docker compose. You should be able to use the CLI. `budi hosting --stop` and then `budi hosting --start` will restart your main container. Or you should be able to do `docker --down` and then `docker --up`.

**Reply by koneubauer** [reply]
We tried restarting the containers. The error message still shows. However, I want to mention once again: The error we get now is different to the one before. Now we get "Failed to cancel request in 5000ms". Before we got "Request failed to complete in 15000ms"

**Reply by koneubauer** [reply]
By now we are on 2.9.34 and the issue still exists.

**Reply by koneubauer** [reply]
Hey @ConorWebb96,
I upgraded to 2.9.38 now and the issue still exists. The error message does not appear after 5 seconds, it takes a bit for it to come up (roughly 20 seconds). Is there any more information that I can give you?

**Reply by ConorWebb96** [reply]
Hey @koneubauer,

I haven't been able to replicate the error you are receiving.

Theres a wide range of things that could be happening here. This could be related to the API you are using. It may also be related to your self-host installation.

Below are the settings using;
```
self-host - Docker compose
Budibase version - v2.9.38
.env - QUERY_THREAD_TIMEOUT=60000
docker-compose.yaml - is shown in the above screenshot
```
I've tested again today and the increased timeout works across the CLI, single image, docker compose and the dev stack.

I hope this helps you.

**Reply by jakobwarnhjelm** [reply]
We've also started to seeing this error message after upgrading to the latest version yesterdag. The previous version was from june/may 2023.
No change in apps.
Setting` QUERY_THREAD_TIMEOUT=60000` and running  
`docker-compose down`
followed by 
`docker-compose up -d --build --force-recreate`
wasn't enough to resolve the issue.

**Reply by tw02d008** [reply]
I am having issues with the QUERY_THREAD_TIMEOUT as well. I am running in the QueryError: Request failed to complete in 15000ms.
I updated my docker compose file an added the QUERY_THREAD_TIMEOUT value with 60000 but still I get the 15000ms timeout error, If i lower the QUERY_THREAD_TIMEOUT to below 15000 the change of the environment variable is reflected in the error message.
It looks like the QUERY_THREAD_TIMEOUT is somehow hardcoded to a maximum of 15000ms? Could it be the case? Anyhow I would love to change it to higher number and not just to a lower number.
I am using selfhosted budibase version 2.24.1


using portainer and updated the stacupdated my

**Reply by Abd-Atif** [reply]
@koneubauer pls try to ping that endpoint first, if it works then add some delay. Afterwards, restart your budibase containers and try !

**Reply by A3DataSolutions** [reply]
Bump, also experiencing the same issue with a SQL Server back end. On the latest budibase version
